pkg/storage/unified/apistore/store.go

```diff
@@ -460,11 +460,13 @@ func (s *Storage) GuaranteedUpdate(
 			if err != nil {
 				return err
 			}
+
 			mmm, err := utils.MetaAccessor(existingObj)
 			if err != nil {
 				return err
 			}
 			mmm.SetResourceVersionInt64(rsp.ResourceVersion)
+			res.ResourceVersion = uint64(rsp.ResourceVersion)
```

Comment:
Is this needed ?

Reply:
Ok. Makes total sense !

Reply:
This one is fascinating. It's insane that it worked without it before. Good catch !